### PR TITLE
BTA-1071 Display VAT Direct Debit information on VAT homepage card

### PR DESCRIPTION
**Pull request by gianick**

### CODE REVIEW

File: app/services/VatPartialBuilder.scala [hunk 1]
```diff
@@ -48,8 +48,14 @@ class VatPartialBuilderImpl @Inject() (val enrolmentsStore: EnrolmentsStoreServi
       case VatData(accountSummaryData, calendar) => accountSummaryData match {
```
Comment (chris-ianson) [inline app/services/VatPartialBuilder.scala]: This file has some warnings (length exceeds 160, cyclomatic complexity) Not major but for me the full file is highlighted.

File: test/services/VatPartialBuilderSpec.scala [hunk 2]
```diff
@@ -239,6 +258,61 @@ class VatPartialBuilderSpec extends ViewSpecBase with OneAppPerSuite with Mockit
...
+          doc,
+          linkId = "vat-make-payment-link",
+          expectedText = "Make a VAT payment",
+          expectedUrl = "http://localhost:9732/business-account/vat/make-a-payment",
+          expectedGAEvent = "link - click:VAT cards:Make a VAT payment",
+          expectedIsExternal = false,
+          expectedOpensInNewTab = false
+        )
+      }
+
```
Comment (AndrewShanks) [inline test/services/VatPartialBuilderSpec.scala]: I'd like to see unit tests for the inactive DD and ineligible DD (for reasons other than annual filing) cases

Comment (gianick) [reply test/services/VatPartialBuilderSpec.scala]: There's already a test for InactiveDirectDebit. I added one for Ineligible.

File: test/services/VatPartialBuilderSpec.scala [hunk 3]
```diff
@@ -79,21 +79,45 @@ class VatPartialBuilderSpec extends ViewSpecBase with OneAppPerSuite with Mockit
...
+      directDebit = activeDirectDebit
+    )
+
+    lazy val calendarWithAnnualFiling: Calendar = Calendar(
+      filingFrequency = Annually,
+      directDebit = activeDirectDebit
+    )
+
+    lazy val calendarWithIneligibilityForDirectDebit: Calendar = Calendar(
+      filingFrequency = Monthly,
```
Comment (spattisson) [inline test/services/VatPartialBuilderSpec.scala]:
```suggestion
      filingFrequency = InvalidStaggerCode,
```

**Pull request merged**